deepspeed-mii/model/model.py

```diff
@@ -0,0 +1,40 @@
+import asyncio
+from typing import Dict
+
+import mii
+from huggingface_hub import login
+
+DEFAULT_RESPONSE_MAX_LENGTH = 512
+
+
+class Model:
+    def __init__(self, **kwargs) -> None:
+        self.hf_access_token = kwargs["secrets"]["hf_access_token"]
+        model_metadata = kwargs["config"]["model_metadata"]
+        self.repo_id = model_metadata["repo_id"]
+        self.max_length = int(model_metadata["max_length"])
+        self.tensor_parallel = int(model_metadata["tensor_parallel"])
+
+    def load(self):
+        login(token=self.hf_access_token)
+        # need to create a new loop because `mii.serve` creates async client at the end,
+        # and `load` function being called from new thread
+        asyncio.set_event_loop(asyncio.new_event_loop())
```

Comment:
Not a blocker, but I imagine there should be a way of accessing the main event loop (that we use in TrussServer) from here instead of creating a new one on this thread. cc @squidarth

Reply:
We investigated it a bit, truss server executes this code from new thread and new threads in python doesn't have eventloops by default. Hence I don't think we really need to utilize main event loop here.